From: Liu Bo <bo.liu@linux.alibaba.com>
Date: Tue, 14 Aug 2018 10:46:53 +0800
Patch-mainline: 4.20
Git-commit: 4b6f8e9695da65e29f9f8ee84b39e0ba5b45e8e9
Subject: [PATCH 09/11] Btrfs: do not unnecessarily pass write_lock_level when
 processing leaf
References: bsc#1126802

As we're going to return right after the call, it's not necessary to get
update the new write_lock_level from unlock_up.

Signed-off-by: Liu Bo <bo.liu@linux.alibaba.com>
Reviewed-by: David Sterba <dsterba@suse.com>
Signed-off-by: David Sterba <dsterba@suse.com>
Signed-off-by: Filipe Manana <fdmanana@suse.com>
---
 fs/btrfs/ctree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/btrfs/ctree.c b/fs/btrfs/ctree.c
index eb73edd75931..cc01890970a8 100644
--- a/fs/btrfs/ctree.c
+++ b/fs/btrfs/ctree.c
@@ -3001,7 +3001,7 @@ int btrfs_search_slot(struct btrfs_trans_handle *trans, struct btrfs_root *root,
 			}
 			if (!p->search_for_split)
 				unlock_up(p, level, lowest_unlock,
-					  min_write_lock_level, &write_lock_level);
+					  min_write_lock_level, NULL);
 			goto done;
 		}
 	}
-- 
2.19.0

